app/checks/free_ram.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python3
+
+import re
+from os import environ
+from sys import stderr
+
+if 'RAMPERCENT' not in environ:
+    print("Missing RAMPERCENT parameter.\nUsage ./free_ram.py env variables are RAMPERCENT, [MOCK_FREE_RAM] (in MiB)", file=stderr)
+    exit(127)
+
+def check_ram(meminfopath, maxPercent):
+    meminfo = open(meminfopath).read()
+    memTotalreg = re.compile("MemTotal:\s+(\d+)")
+    memAvailreg = re.compile("MemAvailable:\s+(\d+)")
+    memTotal = float( memTotalreg.split(meminfo)[1] )
+    memAvail = float( memAvailreg.split(meminfo)[1] )
+    if "MOCK_FREE_RAM" in environ and "MOCK_TOTAL_RAM" in environ:
+        memAvail = float(environ["MOCK_FREE_RAM"]) * 1024
+        memTotal = float(environ["MOCK_TOTAL_RAM"]) * 1024
+    memUsed = memTotal - memAvail
+    memUsedPercent = memUsed/memTotal * 100
+    if memUsed/memTotal * 100 > maxPercent:
+        print("RAM usage too high: {0:.0f}MiB/{1:.0f}MiB ({2:.0f}% used, {3:.0f}MiB left. Max usage allowed: {4:.0f}% ({5:.0f}MiB)".format(memUsed/1024, memTotal/1024, memUsedPercent, memAvail/1024, maxPercent, maxPercent/100.0 * memTotal/1024 ))
+        return 1
+    print("RAM usage OK: {0:.0f}MiB/{1:.0f}MiB ({2:.0f}% used, {3:.0f}MiB left. Max usage allowed: {4:.0f}% ({5:.0f}MiB)".format(memUsed/1024, memTotal/1024, memUsedPercent, memAvail/1024, maxPercent, maxPercent/100.0 * memTotal/1024 ))
+    return 0
+
+
+exit(check_ram("/proc/meminfo", float(environ["RAMPERCENT"])))
```

Comment:
Will work only on Linux, but I don't know if we are actually supporting other platforms.

I see that there is a library for Python, but it would be introducing an additional dependency just to use it once...
https://stackoverflow.com/questions/276052/how-to-get-current-cpu-and-ram-usage-in-python/2468983

So, for now I think that's ok :wink: